src/app/clusters/operational-state-server/operational-state-delegate.h

```diff
@@ -0,0 +1,393 @@
+            {
+                memcpy(ErrorStateDetails, details, kOperationalErrorDetailsMaxSize);
+            }
+            else
+            {
+                memcpy(ErrorStateDetails, details, detailsLen);
+            }
+
+            errorStateDetails.SetValue(chip::CharSpan(ErrorStateDetails, sizeof(ErrorStateDetails)));
+        }
+    }
+    uint8_t getStateID() const { return errorStateID; }
+    char ErrorStateLabel[kOperationalErrorLabelMaxSize];
+    char ErrorStateDetails[kOperationalErrorDetailsMaxSize];
+};
+
+/**
+ * A class presents the phase of operational state cluster
+ */
+struct GenericOperationalPhase
+{
+    char PhaseName[kOperationalPhaseNameMaxSize];
+    chip::app::DataModel::Nullable<chip::CharSpan> phaseName;
+
+    GenericOperationalPhase(const char * name = nullptr, size_t nameLen = 0) { set(name, nameLen); }
+
+    GenericOperationalPhase(const GenericOperationalPhase & ph)
+    {
+        if (ph.isNullable())
+        {
+            set();
+        }
+        else
+        {
+            set(ph.PhaseName, sizeof(ph.PhaseName));
+        }
+    }
+
+    GenericOperationalPhase & operator=(const GenericOperationalPhase & ph)
+    {
+        if (ph.isNullable())
+        {
+            set();
+        }
+        else
+        {
+            set(ph.PhaseName, sizeof(ph.PhaseName));
+        }
+        return *this;
+    }
+
+    void set(const char * name = nullptr, size_t nameLen = 0)
+    {
+        if (name == nullptr)
+        {
+            phaseName.SetNull();
+        }
+        else
+        {
+            memset(PhaseName, 0, sizeof(PhaseName));
+            if (nameLen > kOperationalPhaseNameMaxSize)
+            {
+                memcpy(PhaseName, name, kOperationalPhaseNameMaxSize);
+            }
+            else
+            {
+                memcpy(PhaseName, name, nameLen);
+            }
+            phaseName = chip::app::DataModel::Nullable<chip::CharSpan>(chip::CharSpan(PhaseName, sizeof(PhaseName)));
+        }
+    }
+    bool isNullable() const { return phaseName.IsNull(); }
+};
+
+/**
+ * A class hold the phase list of operational state cluster
+ */
+struct GenericOperationalPhaseList : public GenericOperationalPhase
+{
+    GenericOperationalPhaseList(const char * name = nullptr, size_t nameLen = 0) : GenericOperationalPhase(name, nameLen) {}
+    GenericOperationalPhaseList * next = nullptr;
+};
+
+/**
+ * A class wrap the operation completion of operational state cluster
+ */
+struct GenericOperationCompletion : public chip::app::Clusters::OperationalState::Events::OperationCompletion::Type
+{
+    GenericOperationCompletion(uint8_t aCompletionErrorCode)
+    {
+
+        chip::app::DataModel::Nullable<uint32_t> __totalOperationalTime;
+        chip::app::DataModel::Nullable<uint32_t> __pausedTime;
+
+        completionErrorCode = aCompletionErrorCode;
+
+        __totalOperationalTime.SetNull();
+        totalOperationalTime.SetValue(__totalOperationalTime);
```

Comment:
```suggestion
        totalOperationalTime.SetValue().SetNull();
```

Reply:
`totalOperationalTime.SetValue().SetNull(); ` is a error way ,  use `totalOperationalTime.SetValue(app::DataModel::Nullable<uint32_t>());` instead.